src/components/ingredients/IngredientBoughtOften.tsx

```diff
@@ -0,0 +1,160 @@
+import {
+  Editable,
+  EditableInput,
+  EditablePreview,
+  Flex,
+  Input,
+  Stack,
+  Text,
+} from "@chakra-ui/react";
+import Ingredient from "../recipes/Ingredient.tsx";
+import PlusMinus from "../PlusMinus.tsx";
+import { useMyStore } from "../store/store.tsx";
+import { useFindProductInDelivered } from "../hooks/useFindProductInDelivered.ts";
+import { Dispatch, FC, SetStateAction, useEffect, useState } from "react";
+import { NewIngredient } from "../types.ts";
+import { useFindIngredientById } from "../hooks/useFindIngredientById.ts";
+
+type Props = {
+  ingredient: { id: NewIngredient["id"]; amount: number; frequency: number };
+  setDisable: Dispatch<SetStateAction<boolean>>;
+};
+
+export const IngredientBoughtOften: FC<Props> = ({
+  ingredient,
+  setDisable,
+}) => {
+  const { selectedBoughtOften, editSelectedBoughtOften } = useMyStore();
+  const [error, setError] = useState(ingredient.frequency === 0);
+  const matchingIngredient = useFindIngredientById(ingredient);
+  if (!matchingIngredient) {
+    throw new Error("No matching ingredient found");
+  }
+  const productIds = matchingIngredient.selectedProducts.map(
+    (product) => product.id,
+  );
+  const foundPurchase = useFindProductInDelivered(productIds);
+  console.log({ foundPurchase });
+  const handleDelete = (ingredientId: string) => {
+    const updatedIngredients = selectedBoughtOften.filter(
+      (ingredient) => ingredient.id !== ingredientId,
+    );
+    editSelectedBoughtOften(updatedIngredients);
+  };
+
+  const onSubmit = (nextValue: string) => {
+    console.log(nextValue, "ahoj");
+    if (!nextValue || Number(nextValue) <= 0) {
+      setError(true);
+      return;
+    }
+    setError(false);
+    const updatedIngredients = selectedBoughtOften.map((existingIngredient) => {
+      if (existingIngredient.id === ingredient.id) {
+        return {
+          ...existingIngredient,
+          frequency: Number(nextValue),
+        };
+      }
+      return existingIngredient;
+    });
+    editSelectedBoughtOften(updatedIngredients);
+  };
+
+  const handleAdd = () => {
+    const updatedIngredients = selectedBoughtOften.map((existingIngredient) => {
+      if (existingIngredient.id === ingredient.id) {
+        return {
+          ...existingIngredient,
+          amount: existingIngredient.amount + 1,
+        };
+      }
+      return existingIngredient;
+    });
+    editSelectedBoughtOften(updatedIngredients);
```

Comment:
Tohle IMHO nepatri do UI vrstvy a melo by to byt reseno na urovni store a ala react actions.